### PR TITLE
chore(oas): replace requestBody with $ref to req class JSDoc OAS

### PR DESCRIPTION
**Pull request by patrick-medusajs**
### What

Move inline OAS requestBody schema declaration under their respective class-validator classes in order to expose them through  `#/components/schemas`. Replace inline OAS requestBody schema with a `$ref` reference pointing to the newly declared schema.

### Why

Having requestBody declared as its own "named" schema will allow OAS code generators to output typed entities/DTO that can be manipulate without having to reference the route/operation.

### How

Declare a new @schema JSDoc for each class-validator used to parse and validate request body. Move the current inline requestBody to the new @schema.

### Test

- Ran OAS validator.
- Ran docs build script.

Expect no visible changes to the documentation.

### Out-of-scope

requestBody of type `multipart/form-data` used for file uploads. These will be address as part of CORE-934
- [create-upload.ts](https://github.com/medusajs/medusa/blob/58d23a7b45b6dde8bdeed0bb268139a0017f1649/packages/medusa/src/api/routes/admin/uploads/create-upload.ts#L87-L90)
- [create-protected-upload.ts](https://github.com/medusajs/medusa/blob/58d23a7b45b6dde8bdeed0bb268139a0017f1649/packages/medusa/src/api/routes/admin/uploads/create-protected-upload.ts#L87-L90)

Path Parameter and Query Parameter. These will need more research and experimentation, part of CORE-931

---

Resolves CORE-853

### CODE REVIEW

File: packages/medusa/src/api/routes/admin/publishable-api-keys/delete-channels-batch.ts [hunk 1]
```diff
@@ -116,6 +103,24 @@ export default async (req: Request, res: Response): Promise<void> => {
   res.status(200).json({ publishable_api_key: publishableApiKey })
 }
 
+/**
+ * @schema AdminDeletePublishableApiKeySalesChannelsBatchReq
+ * type: object
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/admin/publishable-api-keys/delete-channels-batch.ts:108]: Added missing `type: object`

File: packages/medusa/src/api/routes/store/orders/confirm-order-request.ts [hunk 2]
```diff
@@ -110,6 +105,16 @@ export default async (req, res) => {
   res.sendStatus(200)
 }
 
+/**
+ * @schema StorePostCustomersCustomerAcceptClaimReq
+ * type: object
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/store/orders/confirm-order-request.ts:110]: Added missing `type: object`

File: packages/medusa/src/api/routes/store/orders/request-order.ts [hunk 3]
```diff
@@ -125,6 +118,18 @@ export default async (req, res) => {
   res.sendStatus(200)
 }
 
+/**
+ * @schema StorePostCustomersCustomerOrderClaimReq
+ * type: object
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/store/orders/request-order.ts:123]: Added missing `type: object`

File: packages/medusa/src/api/routes/store/payment-collections/authorize-batch-payment-sessions.ts [hunk 4]
```diff
@@ -80,6 +75,18 @@ export default async (req, res) => {
   res.status(207).json({ payment_collection })
 }
 
+/**
+ * @schema StorePostPaymentCollectionsBatchSessionsAuthorizeReq
+ * type: object
+ * required:
+ *   - session_ids
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/store/payment-collections/authorize-batch-payment-sessions.ts:82]: Added missing `type: object`
Mark `session_ids` as required

File: packages/medusa/src/api/routes/store/payment-collections/manage-batch-payment-sessions.ts [hunk 5]
```diff
@@ -136,6 +119,30 @@ export class StorePostPaymentCollectionsSessionsReq {
   session_id?: string
 }
 
+/**
+ * @schema StorePostPaymentCollectionsBatchSessionsReq
+ * type: object
+ * required:
+ *   - sessions
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/store/payment-collections/manage-batch-payment-sessions.ts:126]: Added missing `type: object`
Mark `sessions` as required

File: packages/medusa/src/api/routes/store/payment-collections/manage-payment-session.ts [hunk 6]
```diff
@@ -91,6 +86,16 @@ export default async (req, res) => {
   res.status(200).json({ payment_collection: paymentCollection })
 }
 
+/**
+ * @schema StorePaymentCollectionSessionsReq
+ * type: object
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/store/payment-collections/manage-payment-session.ts:91]: Added missing `type: object`

File: packages/medusa/src/api/routes/store/payment-collections/refresh-payment-session.ts [hunk 7]
```diff
@@ -10,21 +10,6 @@ import { PaymentCollectionService } from "../../../../services"
...
- *         required:
- *           - provider_id
- *           - customer_id
- *         properties:
- *           provider_id:
- *             description: The Payment Provider id.
- *             type: string
- *           customer_id:
- *             description: The Customer id.
- *             type: string
```
Comment (patrick-medusajs) [inline packages/medusa/src/api/routes/store/payment-collections/refresh-payment-session.ts:27]: Removed requestBody declaration. Endpoint only expects path parameters.

**Review by patrick-medusajs — COMMENTED** [review]
self-review: highlighting key changes.

**Review by srindom — APPROVED** [review]
LGTM

File: packages/medusa/src/api/routes/admin/customer-groups/add-customers-batch.ts [hunk 8]
```diff
@@ -118,6 +105,23 @@ export default async (req: Request, res: Response) => {
...
+ * required:
+ *   - customer_ids
+ * properties:
+ *   customer_ids:
+ *     description: "The ids of the customers to add"
+ *     type: array
+ *     items:
+ *       required:
+ *         - id
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/customer-groups/add-customers-batch.ts:120]: I remember you created a previous PR regarding adding explicit `type: object`. Should we add it here as well?

File: packages/medusa/src/api/routes/admin/customer-groups/delete-customers-batch.ts [hunk 9]
```diff
@@ -118,6 +105,23 @@ export default async (req: Request, res: Response) => {
...
+ * required:
+ *   - customer_ids
+ * properties:
+ *   customer_ids:
+ *     description: "The ids of the customers to remove"
+ *     type: array
+ *     items:
+ *       required:
+ *         - id
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/customer-groups/delete-customers-batch.ts:120]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/discounts/add-resources-to-condition-batch.ts [hunk 10]
```diff
@@ -124,9 +111,27 @@ export default async (req: Request, res: Response) => {
...
+ * required:
+ *   - resources
+ * properties:
+ *   resources:
+ *     description: The resources to be added to the discount condition
+ *     type: array
+ *     items:
+ *       required:
+ *         - id
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/discounts/add-resources-to-condition-batch.ts:126]: Same comment here regarding `type: object`

**Review by adrien2p — APPROVED** [review]
LGTM

File: packages/medusa/src/api/routes/admin/discounts/delete-resources-from-condition-batch.ts [hunk 11]
```diff
@@ -123,8 +110,26 @@ export default async (req: Request, res: Response) => {
...
+ * required:
+ *   - resources
+ * properties:
+ *   resources:
+ *     description: The resources to be deleted from the discount condition
+ *     type: array
+ *     items:
+ *       required:
+ *         - id
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/discounts/delete-resources-from-condition-batch.ts:128]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/orders/create-claim.ts [hunk 12]
```diff
@@ -394,6 +299,105 @@ export default async (req, res) => {
...
+ *       - replace
+ *       - refund
+ *   claim_items:
+ *     description: The Claim Items that the Claim will consist of.
+ *     type: array
+ *     items:
+ *       required:
+ *         - item_id
+ *         - quantity
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/orders/create-claim.ts:322]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/orders/create-claim.ts [hunk 13]
```diff
@@ -394,6 +299,105 @@ export default async (req, res) => {
...
+ *          type: integer
+ *          description: The price to charge for the Shipping Method.
+ *   additional_items:
+ *      description: The new items to send to the Customer when the Claim type is Replace.
+ *      type: array
+ *      items:
+ *        required:
+ *          - variant_id
+ *          - quantity
+ *        properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/orders/create-claim.ts:366]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/orders/create-claim.ts [hunk 14]
```diff
@@ -394,6 +299,105 @@ export default async (req, res) => {
...
+ *            description: The ID of the Product Variant to ship.
+ *            type: string
+ *          quantity:
+ *            description: The quantity of the Product Variant to ship.
+ *            type: integer
+ *   shipping_methods:
+ *      description: The Shipping Methods to send the additional Line Items with.
+ *      type: array
+ *      items:
+ *         properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/orders/create-claim.ts:377]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/orders/create-fulfillment.ts [hunk 15]
```diff
@@ -140,6 +117,33 @@ export default async (req, res) => {
...
+ *   - items
+ * properties:
+ *   items:
+ *     description: The Line Items to include in the Fulfillment.
+ *     type: array
+ *     items:
+ *       required:
+ *         - item_id
+ *         - quantity
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/orders/create-fulfillment.ts:133]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/orders/create-swap.ts [hunk 16]
```diff
@@ -323,6 +253,80 @@ export default async (req, res) => {
...
+ *   - return_items
+ * properties:
+ *   return_items:
+ *     description: The Line Items to return as part of the Swap.
+ *     type: array
+ *     items:
+ *       required:
+ *         - item_id
+ *         - quantity
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/orders/create-swap.ts:269]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/orders/create-swap.ts [hunk 17]
```diff
@@ -323,6 +253,80 @@ export default async (req, res) => {
...
+ *         type: integer
+ *         description: The price to charge for the Shipping Method.
+ *   additional_items:
+ *     description: The new items to send to the Customer.
+ *     type: array
+ *     items:
+ *       required:
+ *         - variant_id
+ *         - quantity
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/orders/create-swap.ts:301]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/orders/create-swap.ts [hunk 18]
```diff
@@ -323,6 +253,80 @@ export default async (req, res) => {
...
+ *         quantity:
+ *           description: The quantity of the Product Variant to ship.
+ *           type: integer
+ *   custom_shipping_options:
+ *     description: The custom shipping options to potentially create a Shipping Method from.
+ *     type: array
+ *     items:
+ *       required:
+ *         - option_id
+ *         - price
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/orders/create-swap.ts:314]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/orders/request-return.ts [hunk 19]
```diff
@@ -333,6 +287,56 @@ type ReturnObj = {
...
+ *   - items
+ * properties:
+ *   items:
+ *     description: The Line Items that will be returned.
+ *     type: array
+ *     items:
+ *       required:
+ *         - item_id
+ *         - quantity
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/orders/request-return.ts:303]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/orders/update-claim.ts [hunk 20]
```diff
@@ -182,6 +105,87 @@ export default async (req, res) => {
...
+ * properties:
+ *   claim_items:
+ *     description: The Claim Items that the Claim will consist of.
+ *     type: array
+ *     items:
+ *       required:
+ *         - id
+ *         - images
+ *         - tags
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/orders/update-claim.ts:120]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/orders/update-claim.ts [hunk 21]
```diff
@@ -182,6 +105,87 @@ export default async (req, res) => {
...
+ *                 description: Image URL
+ *         metadata:
+ *           description: An optional set of key-value pairs to hold additional information.
+ *           type: object
+ *   shipping_methods:
+ *     description: The Shipping Methods to send the additional Line Items with.
+ *     type: array
+ *     items:
+ *        properties:
+ *          id:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/orders/update-claim.ts:173]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/price-lists/add-prices-batch.ts [hunk 22]
```diff
@@ -142,6 +106,46 @@ export default async (req, res) => {
...
+ * type: object
+ * properties:
+ *   prices:
+ *     description: The prices to update or add.
+ *     type: array
+ *     items:
+ *       required:
+ *         - amount
+ *         - variant_id
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/price-lists/add-prices-batch.ts:120]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/price-lists/create-price-list.ts [hunk 23]
```diff
@@ -193,6 +119,84 @@ class CustomerGroup {
...
+ *       - active
+ *       - draft
+ *   prices:
+ *      description: The prices of the Price List.
+ *      type: array
+ *      items:
+ *        required:
+ *          - amount
+ *          - variant_id
+ *        properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/price-lists/create-price-list.ts:164]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/price-lists/create-price-list.ts [hunk 24]
```diff
@@ -193,6 +119,84 @@ class CustomerGroup {
...
+ *          max_quantity:
+ *            description: The maximum quantity for which the price will be used.
+ *            type: integer
+ *   customer_groups:
+ *     type: array
+ *     description: A list of customer groups that the Price List applies to.
+ *     items:
+ *       required:
+ *         - id
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/price-lists/create-price-list.ts:192]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/price-lists/update-price-list.ts [hunk 25]
```diff
@@ -187,6 +115,82 @@ class CustomerGroup {
...
+ *      - active
+ *      - draft
+ *   prices:
+ *     description: The prices of the Price List.
+ *     type: array
+ *     items:
+ *       required:
+ *         - amount
+ *         - variant_id
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/price-lists/update-price-list.ts:155]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/price-lists/update-price-list.ts [hunk 26]
```diff
@@ -187,6 +115,82 @@ class CustomerGroup {
...
+ *         max_quantity:
+ *           description: The maximum quantity for which the price will be used.
+ *           type: integer
+ *   customer_groups:
+ *     type: array
+ *     description: A list of customer groups that the Price List applies to.
+ *     items:
+ *       required:
+ *         - id
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/price-lists/update-price-list.ts:186]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/products/create-product.ts [hunk 27]
```diff
@@ -467,6 +264,213 @@ class ProductVariantReq {
...
+ *   collection_id:
+ *     description: The ID of the Collection the Product should belong to.
+ *     type: string
+ *   tags:
+ *     description: Tags to associate the Product with.
+ *     type: array
+ *     items:
+ *       required:
+ *         - value
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/products/create-product.ts:327]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/products/create-product.ts [hunk 28]
```diff
@@ -467,6 +264,213 @@ class ProductVariantReq {
...
+ *         id:
+ *           description: The ID of an existing Sales channel.
+ *           type: string
+ *   options:
+ *     description: The Options that the Product should have. These define on which properties the Product's Product Variants will differ.
+ *     type: array
+ *     items:
+ *       required:
+ *         - title
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/products/create-product.ts:350]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/products/create-product.ts [hunk 29]
```diff
@@ -467,6 +264,213 @@ class ProductVariantReq {
...
+ *         title:
+ *           description: The title to identify the Product Option by.
+ *           type: string
+ *   variants:
+ *     description: A list of Product Variants to create with the Product.
+ *     type: array
+ *     items:
+ *       required:
+ *         - title
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/products/create-product.ts:360]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/products/create-product.ts [hunk 30]
```diff
@@ -467,6 +264,213 @@ class ProductVariantReq {
...
+ *           type: string
+ *         metadata:
+ *           description: An optional set of key-value pairs with additional information.
+ *           type: object
+ *         prices:
+ *           type: array
+ *           items:
+ *             required:
+ *               - amount
+ *             properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/products/create-product.ts:418]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/products/create-product.ts [hunk 31]
```diff
@@ -467,6 +264,213 @@ class ProductVariantReq {
...
+ *                 type: integer
+ *               max_quantity:
+ *                 description: The maximum quantity for which the price will be used.
+ *                 type: integer
+ *         options:
+ *           type: array
+ *           items:
+ *             required:
+ *               - value
+ *             properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/products/create-product.ts:442]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/products/create-variant.ts [hunk 32]
```diff
@@ -239,6 +142,107 @@ class ProductVariantOptionReq {
...
+ *     type: string
+ *   metadata:
+ *     description: An optional set of key-value pairs with additional information.
+ *     type: object
+ *   prices:
+ *     type: array
+ *     items:
+ *       required:
+ *         - amount
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/products/create-variant.ts:210]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/products/create-variant.ts [hunk 33]
```diff
@@ -239,6 +142,107 @@ class ProductVariantOptionReq {
...
+ *         max_quantity:
+ *           description: The maximum quantity for which the price will be used.
+ *           type: integer
+ *   options:
+ *     type: array
+ *     items:
+ *       required:
+ *         - option_id
+ *         - value
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/products/create-variant.ts:238]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/products/update-product.ts [hunk 34]
```diff
@@ -399,6 +210,199 @@ class ProductVariantReq {
...
+ *   collection_id:
+ *     description: The ID of the Collection the Product should belong to.
+ *     type: string
+ *   tags:
+ *     description: Tags to associate the Product with.
+ *     type: array
+ *     items:
+ *       required:
+ *         - value
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/products/update-product.ts:265]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/products/update-product.ts [hunk 35]
```diff
@@ -399,6 +210,199 @@ class ProductVariantReq {
...
+ *         value:
+ *           description: The value of the Tag, these will be upserted.
+ *           type: string
+ *   sales_channels:
+ *     description: "[EXPERIMENTAL] Sales channels to associate the Product with."
+ *     type: array
+ *     items:
+ *       required:
+ *         - id
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/products/update-product.ts:278]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/products/update-product.ts [hunk 36]
```diff
@@ -399,6 +210,199 @@ class ProductVariantReq {
...
+ *         id:
+ *           description: The ID of an existing Sales channel.
+ *           type: string
+ *   variants:
+ *     description: A list of Product Variants to create with the Product.
+ *     type: array
+ *     items:
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/products/update-product.ts:286]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/products/update-product.ts [hunk 37]
```diff
@@ -399,6 +210,199 @@ class ProductVariantReq {
...
+ *               max_quantity:
+ *                 description: The maximum quantity for which the price will be used.
+ *                 type: integer
+ *         options:
+ *           type: array
+ *           items:
+ *             required:
+ *               - option_id
+ *               - value
+ *             properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/products/update-product.ts:374]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/products/update-variant.ts [hunk 38]
```diff
@@ -242,6 +148,104 @@ class ProductVariantOptionReq {
...
+ *     type: string
+ *   metadata:
+ *     description: An optional set of key-value pairs with additional information.
+ *     type: object
+ *   prices:
+ *     type: array
+ *     items:
+ *       required:
+ *         - amount
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/products/update-variant.ts:213]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/shipping-options/create-shipping-option.ts [hunk 39]
```diff
@@ -181,6 +119,72 @@ class OptionRequirement {
...
+ *     description: "The amount to charge for the Shipping Option."
+ *     type: integer
+ *   requirements:
+ *     description: "The requirements that must be satisfied for the Shipping Option to be available."
+ *     type: array
+ *     items:
+ *       required:
+ *         - type
+ *         - amount
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/shipping-options/create-shipping-option.ts:163]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/admin/shipping-options/update-shipping-option.ts [hunk 40]
```diff
@@ -160,6 +122,48 @@ class OptionRequirement {
...
+ *     description: "An optional set of key-value pairs with additional information."
+ *     type: object
+ *   requirements:
+ *     description: "The requirements that must be satisfied for the Shipping Option to be available."
+ *     type: array
+ *     items:
+ *       required:
+ *         - type
+ *         - amount
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/admin/shipping-options/update-shipping-option.ts:150]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/store/carts/create-cart.ts [hunk 41]
```diff
@@ -223,6 +190,43 @@ export class Item {
...
+ *      url: https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2#Officially_assigned_code_elements
+ *      description: See a list of codes.
+ *   items:
+ *     description: "An optional array of `variant_id`, `quantity` pairs to generate Line Items from."
+ *     type: array
+ *     items:
+ *       required:
+ *         - variant_id
+ *         - quantity
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/store/carts/create-cart.ts:216]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/store/carts/update-cart.ts [hunk 42]
```diff
@@ -174,6 +114,70 @@ class Discount {
...
+ *         description: A full shipping address object.
+ *       - type: string
+ *         description: The shipping address ID
+ *   gift_cards:
+ *     description: "An array of Gift Card codes to add to the Cart."
+ *     type: array
+ *     items:
+ *       required:
+ *         - code
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/store/carts/update-cart.ts:157]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/store/carts/update-cart.ts [hunk 43]
```diff
@@ -174,6 +114,70 @@ class Discount {
...
+ *         code:
+ *           description: "The code that a Gift Card is identified by."
+ *           type: string
+ *   discounts:
+ *     description: "An array of Discount codes to add to the Cart."
+ *     type: array
+ *     items:
+ *       required:
+ *         - code
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/store/carts/update-cart.ts:167]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/store/payment-collections/manage-batch-payment-sessions.ts [hunk 44]
```diff
@@ -136,6 +119,30 @@ export class StorePostPaymentCollectionsSessionsReq {
...
+ * properties:
+ *   sessions:
+ *     description: "An array of payment sessions related to the Payment Collection. If the session_id is not provided, existing sessions not present will be deleted and the provided ones will be created."
+ *     type: array
+ *     items:
+ *       required:
+ *         - provider_id
+ *         - amount
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/store/payment-collections/manage-batch-payment-sessions.ts:135]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/store/payment-collections/refresh-payment-session.ts (hunk 7 above)
Comment (shahednasser) [reply packages/medusa/src/api/routes/store/payment-collections/refresh-payment-session.ts:27]: I know this is out of the scope of this PR, but after seeing this comment I can also see that in the JS Client code example a `payload` is passed as a third parameter to `refreshPaymentSession`, however looking at this method it actually doesn't accept a payload. We should remove it from the example.

File: packages/medusa/src/api/routes/store/returns/create-return.ts [hunk 45]
```diff
@@ -287,6 +251,46 @@ class Item {
...
+ *     type: string
+ *     description: The ID of the Order to create the Return from.
+ *   items:
+ *     description: "The items to include in the Return."
+ *     type: array
+ *     items:
+ *       required:
+ *         - item_id
+ *         - quantity
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/store/returns/create-return.ts:271]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/store/swaps/create-swap.ts [hunk 46]
```diff
@@ -327,6 +282,55 @@ class AdditionalItem {
...
+ *     type: string
+ *     description: The ID of the Order to create the Swap for.
+ *   return_items:
+ *     description: "The items to include in the Return."
+ *     type: array
+ *     items:
+ *       required:
+ *         - item_id
+ *         - quantity
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/store/swaps/create-swap.ts:303]: Same comment here regarding `type: object`

File: packages/medusa/src/api/routes/store/swaps/create-swap.ts [hunk 47]
```diff
@@ -327,6 +282,55 @@ class AdditionalItem {
...
+ *     type: string
+ *     description: The ID of the Shipping Option to create the Shipping Method from.
+ *   additional_items:
+ *     description: "The items to exchange the returned items to."
+ *     type: array
+ *     items:
+ *       required:
+ *         - variant_id
+ *         - quantity
+ *       properties:
```
Comment (shahednasser) [inline packages/medusa/src/api/routes/store/swaps/create-swap.ts:326]: Same comment here regarding `type: object`

**Review by shahednasser — CHANGES_REQUESTED** [review]
It all looks good to me! I just had comments regarding missing `type: object` across some files.

**Comment by patrick-medusajs** [comment]
@shahednasser excellent point about adding `type: object` to arrays of object. That being said, I would recommend that we do it as a separate PR. This way, we could also fix @schemas under /models. Wdyt?

File: packages/medusa/src/api/routes/store/payment-collections/refresh-payment-session.ts (hunk 7 above)
Comment (patrick-medusajs) [reply packages/medusa/src/api/routes/store/payment-collections/refresh-payment-session.ts:27]: Fixed in 57840eec970ff0811d63bf6d8f991587fd14dd48

**Comment by shahednasser** [comment]
Sure I'm fine with that 👍🏻

**Review by shahednasser — APPROVED** [review]

**Pull request merged**